project/src/components/history-route/history-router.tsx

```diff
@@ -0,0 +1,35 @@
+import {useState, useLayoutEffect} from 'react';
+import {Router} from 'react-router-dom';
+import type {BrowserHistory} from 'history';
+
+export interface HistoryRouterProps {
+  history: BrowserHistory;
+  basename?: string;
```

Comment:
```suggestion
  basename?: string;
```
что значит вопросик в типах?

Reply:
? позволяет использовать доп тип undefind